### PR TITLE
Port ConvertSwitchStatementToExpression analyzer/fixer/tests to share…

### PR DESCRIPTION
**Pull request by mavasani**
…d layer

Straightforward file and resource moves to shared layer. Recommend reviewing with xlf file filter applied: **[Review link](https://github.com/dotnet/roslyn/pull/42171/files?file-filters%5B%5D=.cs&file-filters%5B%5D=.projitems&file-filters%5B%5D=.resx)**

### CODE REVIEW

File: src/Analyzers/CSharp/Analyzers/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -85,9 +89,6 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
                     .Add(Constants.ShouldRemoveNextStatementKey, shouldRemoveNextStatement.ToString(CultureInfo.InvariantCulture))));
         }
 
-        public override bool OpenFileOnly(OptionSet options)
```
Comment (mavasani) [inline src/Analyzers/CSharp/Analyzers/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionDiagnosticAnalyzer.cs]: Redundant override as the base implementation also returns false.

File: src/Analyzers/CSharp/Tests/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs [hunk 2]
```diff
@@ -16,6 +14,14 @@
     Microsoft.CodeAnalysis.CSharp.ConvertSwitchStatementToExpression.ConvertSwitchStatementToExpressionDiagnosticAnalyzer,
     Microsoft.CodeAnalysis.CSharp.ConvertSwitchStatementToExpression.ConvertSwitchStatementToExpressionCodeFixProvider>;
 
+#if CODE_STYLE
+using Microsoft.CodeAnalysis.CSharp.Internal.CodeStyle;
+using Microsoft.CodeAnalysis.Internal.Options;
+#else
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+#endif
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/Tests/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs]: these will go away with @sharwell 's option work right?

Comment (mavasani) [reply src/Analyzers/CSharp/Tests/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs]: Yep, that is correct.

File: src/Analyzers/CSharp/Analyzers/CSharpAnalyzers.projitems [hunk 3]
```diff
@@ -13,6 +13,9 @@
     <None Include="$(MSBuildThisFileDirectory)CSharpAnalyzersResources.resx" />
   </ItemGroup>
   <ItemGroup>
+    <Compile Include="$(MSBuildThisFileDirectory)ConvertSwitchStatementToExpression\ConvertSwitchStatementToExpressionConstants.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)ConvertSwitchStatementToExpression\ConvertSwitchStatementToExpressionDiagnosticAnalyzer.Analyzer.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)ConvertSwitchStatementToExpression\ConvertSwitchStatementToExpressionDiagnosticAnalyzer.cs" />
```
Comment (alrz) [inline src/Analyzers/CSharp/Analyzers/CSharpAnalyzers.projitems]: @sharwell @mavasani Do we need to do this for any new analyzer? Hopefully not..

Comment (mavasani) [reply src/Analyzers/CSharp/Analyzers/CSharpAnalyzers.projitems]: @tmat is taking care of this in https://github.com/dotnet/roslyn/pull/42517. However, please add new analyzers to this shared project, so they automatically get included in the CodeStyle NuGet package and are available in CI.

Comment (alrz) [reply src/Analyzers/CSharp/Analyzers/CSharpAnalyzers.projitems]: also tagging @CyrusNajmabadi

**Pull request merged**